Automattic/jetpack
# Improve PHP 8 compatibility

#### Changes proposed in this Pull Request:

Fix PHP 8 issues detected by the development version of [PHPCompatibility](https://packagist.org/packages/phpcompatibility/php-compatibility), then add a (temporary) Github Action to run it against future pull requests.

This also refactors the phpcs config to separate out the specification of paths and the specification of rules (because the PHPCompatibility run needs the former but not the latter), and adjusts the `composer php:compatibility` script to also make use of the separated specification of paths rather than using its own exclude list. And then it fixes compatibility in some files that had been falling through the cracks, and 1000 (hyperbole) other failures that phpcs then whines about since the files are being touched.

#### Jetpack product discussion


See #17166.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* See that the new Github Action runs and passes.

* Run `composer php:compatibility` and see that it runs on all the same files that e.g. `composer php:lint` runs on.

#### Proposed changelog entry for your changes:

* Developer only, none needed.
